src/platforms/x11/input/input_device.cpp

```diff
@@ -201,24 +193,45 @@ void mix::XInputDevice::pointer_release(
             button_state,
             pointer_pos.x.as_value(),
             pointer_pos.y.as_value(),
-            scroll.dx.as_value(),
-            scroll.dy.as_value(),
+            0,
+            0,
             movement.dx.as_value(),
             movement.dy.as_value()
             )
         );
 
 }
 
-void mix::XInputDevice::pointer_motion(
+void mix::XInputDevice::pointer_motion(EventTime event_time, mir::geometry::PointF pos)
```

Comment:
It looks to me like code duplication could be removed by this function delegating to `pointer_axis_motion()` with `mir_pointer_axis_source_none` and `0, 0` for scroll.

Perhaps this function could even be removed and callers could just send those arguments to `pointer_axis_motion()`? Don't feel strongly about that either way.

Reply:
I don't see how this suggestion improves anything. (At least not without a wider rework of the event builder stack which deserves its own PR!)